### PR TITLE
better-Graphic

### PR DESCRIPTION
**Pull request by JanKocabek**
changing graphic from flat rectangle to 2.5D blocks

## Summary by Sourcery

Update rendering system to use 3D-styled block graphics and refactor board representation to use TetrominoType instead of BlockContent.

New Features:
- Introduce a BlockGraphic system with Renderable shapes, sides, and color palettes to render beveled 2.5D tetromino blocks.
- Add TetrominoType enum as the canonical representation of tetromino shapes, including a NON value for empty cells.

Enhancements:
- Refactor GameBoard and BoardView to store and expose TetrominoType values instead of BlockContent, including test utilities and line-clearing logic.
- Update TetrisDrawingHandler and TetrisCanvas to use the new block rendering pipeline, fast rendering hints, and a darker background color.
- Adjust game parameters such as block size and pixel coordinates to match the new block graphics and movement grid.
- Annotate drawing methods with nullability and improve internal APIs like repaintCanvas for clearer dirty-flag semantics.

Build:
- Rename Maven artifact to SwingTetris, bump version to 0.10.5, configure explicit Java 21 source/target, and add jspecify dependency.

Tests:
- Update and expand GameBoard tests to reflect TetrominoType-based board contents, new pixel coordinates, rotation behavior, and additional edge cases for movement and rotation logic.

### CODE REVIEW

File: src/test/java/org/sehes/tetris/model/GameBoardDiffblueTest.java [hunk 1]
```diff
@@ -491,7 +564,7 @@ void testTryRotatePiece_thenGameBoardCurrentTetrominoStateCordSecondYIsMinusOne(
         assertEquals(2, getResult3.y());
         assertEquals(Orientation.WEST, currentTetromino.getCurrentOrientation());
         assertArrayEquals(
```
[the remark this answers is not in the document]
Comment (JanKocabek) [reply src/test/java/org/sehes/tetris/model/GameBoardDiffblueTest.java:566]: @sourcery-ai make a issue

**Comment by JanKocabek** [comment]
@sourcery-ai can you make a new review if there is something other to do?:

File: src/main/java/org/sehes/tetris/config/GameParameters.java [hunk 2]
```diff
@@ -11,16 +11,12 @@
...
+public record GameParameters() {
+    public static final int BLOCK_SIZE = 40;//size of one cell of GameBoard grid, also the size of one block of tetromino
     public static final int COLUMNS = 10;//number of cell in row (columns)
     public static final int ROWS = 22;//number of cell in column (rows)
     public static final int VISIBLE_ROWS = 20;//number of rows that are visible to player, the top 2 rows are for spawning new tetromino and are not visible to player
-    public static final int GAME_SPEED = 1600;//the speed of the game, in milliseconds, the lower the faster
+    public static final int GAME_SPEED = 1600;//the speed of the game, in milliseconds, the lower is the value the faster is the movement
     public static final int HIDDEN_ROWS = ROWS - VISIBLE_ROWS;//the number of rows that are not visible to player, used for calculating the position of tetromino on the game board
     public static final Coordinate SPAWN_POINT = new Coordinate(COLUMNS / 2 - 1, 1);//the point where new tetromino spawns, it is in the middle of the top row of the game board
```
[the remark this answers is not in the document]
Comment (JanKocabek) [reply src/main/java/org/sehes/tetris/config/GameParameters.java]: reworked from record to final class in [5b79ea3](https://github.com/JanKocabek/JavaSwingTetris/pull/27/commits/5b79ea326460d62a3eae75c81a5415675ca77365)

File: src/test/java/org/sehes/tetris/model/GameBoardDiffblueTest.java [hunk 3]
```diff
@@ -158,71 +153,43 @@ void testTryMovePiece_thenGameBoardCurrentTetrominoPositionYIsMinusOne() {
         assertEquals(-1, currentTetromino.getPositionY());
         assertFalse(actualTryMovePieceResult);
         assertArrayEquals(
-                new int[]{-30, -90, 0, -90, -60, -90, 30, -90}, currentTetromino.getPixelCoordinates());
+                new int[]{-40, -120, 0, -120, -80, -120, 40, -120},
```
[the remark this answers is not in the document]
Comment (JanKocabek) [reply src/test/java/org/sehes/tetris/model/GameBoardDiffblueTest.java:156]: @SOURCERY-AI make a issue

File: src/test/java/org/sehes/tetris/model/GameBoardDiffblueTest.java [hunk 4]
```diff
@@ -448,7 +484,44 @@ void testTryRotatePiece_thenGameBoardCurrentTetrominoStateCordFirstXIsOne() {
...
+    void testTryRotatePiece_whenNull() {
+        // Arrange
+        GameBoard gameBoard = new GameBoard();
+        gameBoard.spawnTetrominoForTestOnly(
+                TetrominoFactory.spawnSpecificTetromino(TetrominoType.I, GameParameters.SPAWN_POINT));
+
+        // Act
+        boolean actualTryRotatePieceResult = gameBoard.tryRotatePiece(null);
+
+        // Assert
```
[the remark this answers is not in the document]
Comment (JanKocabek) [reply src/test/java/org/sehes/tetris/model/GameBoardDiffblueTest.java:513]: @SOURCERY-AI make a issue

File: src/main/java/org/sehes/tetris/graphic/ColorPalette.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+     *
+     *
+     */
+
+    static class Colors {
+        private Colors() {}
+        private static float wrapHue(float h) {
+            return (h = h % 1f) < 0 ? h + 1f : h;
+        }
+
```
[the remark this answers is not in the document]
Comment (JanKocabek) [reply src/main/java/org/sehes/tetris/graphic/ColorPalette.java:80]: @SOURCERY-AI make a issue

File: src/main/java/org/sehes/tetris/graphic/ColorPalette.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    public static Paint getPaint(TetrominoType type, Side side) {
+        final var sideMap = COLOR_MAP.get(type);
+        return switch (side) {
+            case TOP -> sideMap[0];
+            case RIGHT -> sideMap[1];
+            case BOTTOM -> sideMap[2];
+            case LEFT -> sideMap[3];
+            case CENTER -> sideMap[4];
```
[the remark this answers is not in the document]
Comment (JanKocabek) [reply src/main/java/org/sehes/tetris/graphic/ColorPalette.java:59]: @SOURCERY-AI make a issue
